web/src/main/java/org/wso2/testgrid/web/api/TestPlanService.java

```diff
@@ -394,4 +401,71 @@ private TestExecutionSummary getTestExecutionSummary(TestPlan testPlan) {
         }
         return new TestExecutionSummary(scenarioSummaries, scenarioTestCaseEntries);
     }
+
+    /**
+     * Returns the result (archive file) to the given scenario of a test plan.
+     *
+     * @param id test plan id
+     * @param scenarioDir scenario directory name
+     * @return archived file of the results
+     */
+    @GET
+    @Path("/result/{id}/{scenarioDir}")
+    public Response getScenarioResult(@PathParam("id") String id, @PathParam("scenarioDir") String scenarioDir) {
+        try {
+            TestPlanUOW testPlanUOW = new TestPlanUOW();
+            Optional<TestPlan> testPlanOptional = testPlanUOW.getTestPlanById(id);
+            if (!testPlanOptional.isPresent()) {
+                return Response.status(Response.Status.NOT_FOUND)
+                        .entity("Couldn't found a TestPlan for the requested id").build();
+            }
+            TestPlan testPlan = testPlanOptional.get();
+            String testPlanDir = TestGridUtil.deriveTestPlanDirName(testPlan);
+            String productName = testPlan.getDeploymentPattern().getProduct().getName();
+            String archiveFileDir = Paths.get(TestGridConstants.TESTGRID_JOB_DIR, productName,
+                    TestGridConstants.TESTGRID_BUILDS_DIR, testPlanDir, scenarioDir,
+                    scenarioDir + TESTGRID_COMPRESSED_FILE_EXT).toString();
+            String bucketKey = Paths
+                    .get(AWS_BUCKET_ARTIFACT_DIR, archiveFileDir).toString();
+            ArtifactReadable artifactDownloadable = new AWSArtifactReader(
+                    ConfigurationContext.getProperty(ConfigurationContext.ConfigurationProperties.AWS_REGION_NAME),
+                    ConfigurationContext.getProperty(ConfigurationContext.ConfigurationProperties.AWS_S3_BUCKET_NAME));
+
+            if (artifactDownloadable.isArtifactExist(bucketKey)) {
+                return Response
+                        .ok(artifactDownloadable.getArtifactStream(bucketKey))
+                        .type(RESPONSE_HEADER_VALUE_APPLICATION_ZIP)
+                        .header(RESPONSE_HEADER_CONTENT_DISPOSITION, RESPONSE_HEADER_VALUE_ATTACHMENT + "; " +
+                                RESPONSE_HEADER_FILE_NAME + "=\" " + scenarioDir +
+                                TESTGRID_COMPRESSED_FILE_EXT + "\"")
+                        .build();
+            } else {
+                return Response.status(Response.Status.NOT_FOUND)
+                        .entity("Couldn't found result-artifacts in the remote location").build();
```

Comment:
At the same time, we should log an error to testgrid.log mentioning this error condition. 
The log should have more context - things like product name, test-plan-id, scenario-id, the s3 result-artifacts location you were trying, etc.